MapboxVisionAR/Render/Renderer/ARRenderer.swift

```diff
@@ -0,0 +1,427 @@
+            let arrowVertexFunction = library.makeFunction(name: ARConstants.ShaderName.arrowVertexMain),
+            let backgroundVertexFunction = library.makeFunction(name: ARConstants.ShaderName.mapTextureVertex),
+            let defaultFragmentFunction = library.makeFunction(name: ARConstants.ShaderName.defaultFragmentMain),
+            let arrowFragmentFunction = library.makeFunction(name: ARConstants.ShaderName.laneFragmentMain),
+            let backgroundFragmentFunction = library.makeFunction(name: ARConstants.ShaderName.displayTextureFragment)
+        else {
+            throw ARRendererError.cantFindFunctions
+        }
+
+        renderPipelineDefault = try ARRenderer.makeRenderPipeline(
+            device: device,
+            vertexDescriptor: vertexDescriptor,
+            vertexFunction: defaultVertexFunction,
+            fragmentFunction: defaultFragmentFunction,
+            colorPixelFormat: colorPixelFormat,
+            depthStencilPixelFormat: depthStencilPixelFormat
+        )
+
+        renderPipelineLane = try ARRenderer.makeRenderPipeline(
+            device: device,
+            vertexDescriptor: vertexDescriptor,
+            vertexFunction: arrowVertexFunction,
+            fragmentFunction: arrowFragmentFunction,
+            colorPixelFormat: colorPixelFormat,
+            depthStencilPixelFormat: depthStencilPixelFormat
+        )
+
+        renderPipelineBackground = try ARRenderer.makeRenderBackgroundPipeline(
+            device: device,
+            vertexDescriptor: ARRenderer.makeTextureMappingVertexDescriptor(),
+            vertexFunction: backgroundVertexFunction,
+            fragmentFunction: backgroundFragmentFunction,
+            colorPixelFormat: colorPixelFormat,
+            depthStencilPixelFormat: depthStencilPixelFormat
+        )
+
+        samplerStateDefault = ARRenderer.makeDefaultSamplerState(device: device)
+        depthStencilStateDefault = ARRenderer.makeDefaultDepthStencilState(device: device)
+
+        guard let buffer = device.makeBuffer(bytes: ARConstants.textureMappingVertices,
+                                             length: ARConstants.textureMappingVertices.count * MemoryLayout<Float>.size,
+                                             options: [])
+        else { throw ARRendererError.cantCreateBuffer }
+        backgroundVertexBuffer = buffer
+
+        super.init()
+    }
+
+    // MARK: Public functions
+
+    func initARSceneForARLane() throws {
+        scene.rootNode.removeAllChildNodes()
+        laneMesh = try self.loadMesh(named: ARConstants.arLaneMeshName)
+        scene.rootNode.add(childNode: ARLaneNode())
+    }
+
+    func set(laneVisualParameters: LaneVisualParams) {
+        if let arLaneNode = scene.arLaneNode() {
+            arLaneNode.set(laneColor: laneVisualParameters.color)
+            arLaneNode.set(laneWidth: laneVisualParameters.width)
+            arLaneNode.set(lightPosition: float3(Float(laneVisualParameters.lightPosition.x),
```

Comment:
here we set light position with values provided by user without converting it, but user assumes that he should provide value in world coordinate system because `LaneVisualParams`'s `lightPosition` property has type `WorldCoordinate`, meanwhile we will treat it as render coordinates. We should convert `lightPosition' value to render coordinate system